### PR TITLE
RDM-13025

### PR DESCRIPTION
**Pull request by KleoG**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/RDM-13025 _"Establish Collection type field for SearchParty "_ (parent ticket)
https://tools.hmcts.net/jira/browse/RDM-13049 _"Update definition store to recognise new column - CollectionFieldName - in SearchParty tab"_ (sub task **under review**)

### Change description ###
Establish Collection type field for SearchParty

> This ticket is dependent on:
> * hmcts/ccd-data-store-api#1828
> * ~hmcts/ccd-test-definitions#32~
> * hmcts/ccd-test-definitions#36

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Review by sem-hmcts — APPROVED** [review]

File: aat/src/aat/resources/features/F-099 SearchParty (GlobalSearch)/F-099.feature [hunk 1]
```diff
@@ -11,11 +11,11 @@ Feature: F-099: Search Party (Global Search)
     And it is submitted to call the [Import definition file] operation of [CCD Definition Store]
     Then a positive response is received
     And the response has all other details as expected
-    And a call [fetch Complex Fields Ordering Schema to verify single case field values in columns SearchPartyName(AC1), SearchPartyEmailAddress(AC6), SearchPartyAddressLine1(AC8), SearchPartyPostCode(AC10) and SearchPartyDoB(AC12)] will get the expected response as in [S-099.1_Get_ComplexFieldsOrdering].
+    And a call [fetch Complex Fields Ordering Schema to verify single case field values in columns SearchPartyName(AC1), SearchPartyEmailAddress(AC6), SearchPartyAddressLine1(AC8), SearchPartyPostCode(AC10), SearchPartyDoB(AC12) and SearchPartyCollectionFieldName] will get the expected response as in [S-099.1_Get_ComplexFieldsOrdering].
```
Comment (mattnayler) [inline aat/src/aat/resources/features/F-099 SearchParty (GlobalSearch)/F-099.feature:14]: Please switch to using a more up to date test definition file as adding `SearchPartyCollectionFieldName` to existing config is unlikely to pass additional validation added in [RDM-13106](https://tools.hmcts.net/jira/browse/RDM-13106).  i.e. reset this check and add a new verify step to check  `SearchPartyCollectionFieldName` using a different case type from the updated def file.

File: excel-importer/src/test/java/uk/gov/hmcts/ccd/definition/store/excel/validation/SearchPartyValidatorTest.java [hunk 2]
```diff
@@ -73,7 +76,7 @@ void setUp() {
     void shouldValidateAllFields() {
 
         // GIVEN
-        SearchPartyEntity searchPartyEntity1 = createPopulatedSearchPartyEntity();
+        SearchPartyEntity searchPartyEntity1 = createPopulatedSearchPartyEntityWithoutCollectionField();
```
Comment (mattnayler) [inline excel-importer/src/test/java/uk/gov/hmcts/ccd/definition/store/excel/validation/SearchPartyValidatorTest.java:79]: Please rename function and DisplayName e.g. "should validate all fields without collection field"

File: excel-importer/src/test/java/uk/gov/hmcts/ccd/definition/store/excel/validation/SearchPartyValidatorTest.java [hunk 3]
```diff
@@ -90,7 +93,31 @@ void shouldValidateAllFields() {
...
         verifyDotNotationValidatorCallMadeFor(ColumnName.SEARCH_PARTY_DOD,
-            searchPartyEntity1.getSearchPartyDod());
+                                              searchPartyEntity1.getSearchPartyDod());
+
+        verifyDotNotationValidatorCallNeverMadeFor(ColumnName.SEARCH_PARTY_COLLECTION_FIELD_NAME);
+    }
+
+    @DisplayName("should validate all fields")
+    @Test
+    void shouldOnlyValidateCollectionField() {
```
Comment (mattnayler) [inline excel-importer/src/test/java/uk/gov/hmcts/ccd/definition/store/excel/validation/SearchPartyValidatorTest.java:103]: Please rename function and DisplayName e.g. "should validate all fields with collection field".

NB: RDM-13106 should fill in the gaps.

**Review by mattnayler — CHANGES_REQUESTED** [review]
Minor change required prior to RDM-13106.

File: excel-importer/src/main/java/uk/gov/hmcts/ccd/definition/store/excel/validation/SearchPartyValidator.java [hunk 4]
```diff
@@ -113,6 +117,21 @@ private void validateSearchPartyDod(ParseContext parseContext,
         }
     }
 
+    private void validateSearchPartyCollectionFieldName(ParseContext parseContext,
+                                         String caseType,
+                                         SearchPartyEntity searchPartyEntity) {
+
+        String spCollectionFieldName = searchPartyEntity.getSearchPartyCollectionFieldName();
+
+        if (StringUtils.isNoneBlank(spCollectionFieldName)) {
```
[the remark this answers is not in the document]
Comment (KleoG) [reply excel-importer/src/main/java/uk/gov/hmcts/ccd/definition/store/excel/validation/SearchPartyValidator.java]: Fixed in [3522292](https://github.com/hmcts/ccd-definition-store-api/pull/1130/commits/3522292204d6d0bc99422bd9abf6c58bbe2cbf8f)

File: aat/src/aat/resources/features/F-099 SearchParty (GlobalSearch)/F-099.feature (hunk 1 above)
Comment (KleoG) [reply aat/src/aat/resources/features/F-099 SearchParty (GlobalSearch)/F-099.feature]: Fixed in [3522292](https://github.com/hmcts/ccd-definition-store-api/pull/1130/commits/3522292204d6d0bc99422bd9abf6c58bbe2cbf8f)

File: excel-importer/src/test/java/uk/gov/hmcts/ccd/definition/store/excel/validation/SearchPartyValidatorTest.java (hunk 3 above)
Comment (KleoG) [reply excel-importer/src/test/java/uk/gov/hmcts/ccd/definition/store/excel/validation/SearchPartyValidatorTest.java]: Fixed in [3522292](https://github.com/hmcts/ccd-definition-store-api/pull/1130/commits/3522292204d6d0bc99422bd9abf6c58bbe2cbf8f)

**Review by mattnayler — APPROVED** [review]
👍  LGTM after recent changes

File: charts/ccd-definition-store-api/values.preview.template.yaml [hunk 5]
```diff
@@ -23,7 +23,7 @@ java:
     ELASTIC_SEARCH_ENABLED: true
 
     # TODO: remove TEMPORARY OVERRIDE FOR GlobalSearch R2, reinstate: ELASTIC_SEARCH_HOST: ccd-data-store-api-pr-1260-es-master
-    ELASTIC_SEARCH_HOST: ccd-data-store-api-pr-1836-es-master
+    ELASTIC_SEARCH_HOST: ccd-data-store-api-pr-1828-es-master
```
Comment (mattnayler) [inline charts/ccd-definition-store-api/values.preview.template.yaml:26]: Temporary values should be updated prior to merge into `Global_Search_Release2` branch and then removed before that branch is merged for release.

**Pull request merged**